families/seth/rpc/src/calls/logs.rs

```diff
@@ -174,20 +190,157 @@ pub fn get_filter_logs<T>(params: Params, mut client: ValidatorClient<T>)
 {
     info!("eth_getFilterLogs");
     let filter_id = to_filter_id(params.parse()?)?;
-
-    let filter = client.get_filter(&filter_id).map_err(Error::invalid_params)?;
-    debug!("filter: {:?}", filter);
-
-    Err(error::not_implemented())
+    let (filter, block_id) = client.get_filter(&filter_id).map_err(Error::invalid_params)?;
+    if let Filter::Log(filter_spec) = filter {
+        get_logs_by_filter(client, Some(filter_id), filter_spec, block_id)
+    } else {
+        Err(Error::invalid_params(format!("Filter {} is not a log filter", filter_id)))
+    }
 }
 
-pub fn get_logs<T>(_params: Params, mut _client: ValidatorClient<T>)
+pub fn get_logs<T>(_params: Params, mut client: ValidatorClient<T>)
     -> Result<Value, Error> where T: MessageSender
 {
     info!("eth_getLogs");
     let params: Value = _params.parse()?;
     let log_filter = LogFilterSpec::from_value(params)?;
-    debug!("filter: {:?}", log_filter);
+    let current_block = client.get_current_block().map_err(|e| {
+        error!("Unable to get current block: {:?}", e);
+        Error::internal_error()
+    })?;
+
+    let block_header: BlockHeader = protobuf::parse_from_bytes(&current_block.header).map_err(|e| {
+        error!("Error parsing block header: {:?}", e);
+        Error::internal_error()
+    })?;
+
+    // Filter from the previous block, so as to find all the logs in this block that match
+    get_logs_by_filter(client, None, log_filter, block_header.previous_block_id)
+}
+
+///
+/// Returns the logs by the given filter, since the last block id
+fn get_logs_by_filter<T>(mut client: ValidatorClient<T>,
+                         _filter_id: Option<String>,
+                         filter_spec: LogFilterSpec,
+                         last_block_id: String)
+    -> Result<Value, Error> where T: MessageSender
+{
+    debug!("log filter: {:?} since {}", filter_spec, &last_block_id[..8]);
+    let mut paging = PagingControls::new();
+    paging.set_count(100);
```

Comment:
Why 100?

Reply:
This is a way to fetch back far enough, with the assumption that this is being called regularly (as it is meant to be polled).  Was a balance between fetching too many blocks.  Will probably change this in a future PR to fetch chunks.  Or to simply store relevant events in map that is updated via subscriptions.